platform/kube/inject/initializer.go

```diff
@@ -0,0 +1,456 @@
+		return err
+	}
+	patchBytes, err := i.createTwoWayMergePatch(in, out, appsv1beta1.Deployment{})
+	if err != nil {
+		return err
+	}
+	_, err = i.clientset.AppsV1beta1().Deployments(in.Namespace).
+		Patch(in.Name, types.StrategicMergePatchType, patchBytes)
+	return err
+}
+
+func (i *Initializer) initializeStatefulSet(obj interface{}) error {
+	in := obj.(*appsv1beta1.StatefulSet)
+	if !i.initializerPresent(&in.ObjectMeta) {
+		return nil
+	}
+
+	glog.Infof("Initializing statefulset: %s", in.Name)
+
+	o, err := runtime.NewScheme().DeepCopy(in)
+	if err != nil {
+		return err
+	}
+	out := o.(*appsv1beta1.StatefulSet)
+
+	if err = i.modifyResource(&out.ObjectMeta, &out.Spec.Template.ObjectMeta, &out.Spec.Template.Spec); err != nil {
+		return err
+	}
+	patchBytes, err := i.createTwoWayMergePatch(in, out, appsv1beta1.StatefulSet{})
+	if err != nil {
+		return err
+	}
+	_, err = i.clientset.AppsV1beta1().StatefulSets(in.Namespace).
+		Patch(in.Name, types.StrategicMergePatchType, patchBytes)
+	return err
+}
+
+func (i *Initializer) initializeJob(obj interface{}) error {
+	in := obj.(*batchv1.Job)
+	if !i.initializerPresent(&in.ObjectMeta) {
+		return nil
+	}
+
+	glog.Infof("Initializing job: %s", in.Name)
+
+	o, err := runtime.NewScheme().DeepCopy(in)
+	if err != nil {
+		return err
+	}
+	out := o.(*batchv1.Job)
+
+	if err = i.modifyResource(&out.ObjectMeta, &out.Spec.Template.ObjectMeta, &out.Spec.Template.Spec); err != nil {
+		return err
+	}
+	patchBytes, err := i.createTwoWayMergePatch(in, out, batchv1.Job{})
+	if err != nil {
+		return err
+	}
+	_, err = i.clientset.BatchV1().Jobs(in.Namespace).
+		Patch(in.Name, types.StrategicMergePatchType, patchBytes)
+	return err
+}
+
+func (i *Initializer) initializeDaemonSet(obj interface{}) error {
+	in := obj.(*v1beta1.DaemonSet)
+	if !i.initializerPresent(&in.ObjectMeta) {
+		return nil
+	}
+
+	glog.Infof("Initializing daemonset: %s", in.Name)
+
+	o, err := runtime.NewScheme().DeepCopy(in)
+	if err != nil {
+		return err
+	}
+	out := o.(*v1beta1.DaemonSet)
+
+	if err = i.modifyResource(&out.ObjectMeta, &out.Spec.Template.ObjectMeta, &out.Spec.Template.Spec); err != nil {
+		return err
+	}
+	patchBytes, err := i.createTwoWayMergePatch(in, out, v1beta1.DaemonSet{})
+	if err != nil {
+		return err
+	}
+	_, err = i.clientset.ExtensionsV1beta1().DaemonSets(in.Namespace).
+		Patch(in.Name, types.StrategicMergePatchType, patchBytes)
+	return err
+}
+
+func (i *Initializer) initializeReplicaSet(obj interface{}) error {
+	in := obj.(*v1beta1.ReplicaSet)
+	if !i.initializerPresent(&in.ObjectMeta) {
+		return nil
+	}
+
+	glog.Infof("Initializing replicaset: %s", in.Name)
+
+	o, err := runtime.NewScheme().DeepCopy(in)
+	if err != nil {
+		return err
+	}
+	out := o.(*v1beta1.ReplicaSet)
+
+	if err = i.modifyResource(&out.ObjectMeta, &out.Spec.Template.ObjectMeta, &out.Spec.Template.Spec); err != nil {
```

Comment:
you could try code generation to get around this repetitive code problem (like what I had to do for CRDs)